src/main/java/dev/streamx/sling/connector/StreamXIngestion.java

```diff
@@ -0,0 +1,35 @@
+package dev.streamx.sling.connector;
+
+import java.util.List;
+
+/**
+ * StreamX REST Ingestion Service.
+ */
+public interface StreamXIngestion {
```

Comment:
This connector is in use already. We should not change the api for no reasons.
I think all the changes from this main package should be reverted and the feature should be implemented as an implementation of RelatedResourcesSelector without any changes of existing code in api or even impl (just new RelatedResourcesSelector impl and related coded added)